### DISCUSSION TITLE
vercel build --dry-run

### DISCUSSION THREAD

**Discussion by switz** [vercel/vercel#8365] [Ideas]
It would be nice to be able to build a project locally without emitting any build files.

Context:

I like running a build before deploying so I get fewer surprises after deploying, but building breaks my dev server. Would be nice to build without actually emitting any code, so the dev server doesn't break and require a restart.

`vercel build --dry-run && git push origin HEAD` would let me ensure the build succeeded before pushing. `--noEmit` could be an alternative flag.

**Comment by amyegan** [comment]
Thanks for sharing your idea @switz! Can you help me understand how the current [`vercel build`](https://vercel.com/docs/cli/build) behavior is breaking your dev server?

**Reply by switz** [reply]
when I run a build it breaks my actively running dev server as well as emits sourcemaps to third parties

**Reply by amyegan** [reply]
The [`build`](https://vercel.com/docs/cli/build) command only builds the project locally or on CI. Nothing should be deployed until you run the `deploy` command. Do you have other scripts set to run at the same time which could be altering your dev server and sourcemaps? 

Please share a more detailed explanation of how your actively running dev server is being broken. It would also help if you can provide a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so we can get a better understanding of what is going wrong for you.
